### ISSUE TITLE
zmcontrol restart stucked after renewing certificate

### ISSUE THREAD

**Issue by CesarCapillas** [VojtechMyslivec/letsencrypt-zimbra#62]
Hi:

First of all, thanks for the project. 
We are using it for renewal Let's Encript certificates in Zimbra 8.8.12 with Ubuntu 16.04.6 LTS, since some months ago.

We are experiencing some kind of issue when renewing certificate (via cron) because, although cert renewal is done correctly, from that moment the mail server is not able to send or receive emails (and all messages are deferred in queue). We simply need to restart the zimbra services (zmcontrol restart), and everything works like a charm, and so the queue starts to work again.

Does it have sense to you ? Is this a necessary condition after renewing certificates ? As far I read, the script restarts the zimbra service.

Kind regards.
--C.

**Comment by VojtechMyslivec** [comment]
Hi,

> First of all, thanks for the project.

you are welcome. Nice to hear that.

> As far I read, the script restarts the zimbra service.

Yes, the script should restart zimbra services when a cert is renewed.

1. What version of letsencrypt-zimbra tool are you using? 
    ```
    /opt/letsencrypt-zimbra/obtain-and-deploy-letsencrypt-cert.sh -V
    ```
2. Is that problem occur only after renewing the cert? Or is it more often?
3. Are you able to receive e-mail notification about failed cron job? If so, the script should report any error via e-mail
4. It is possible to renew the cert manually by this command (do not do this  during rush hours as it will restart zimbra service):
    ```
    sudo -iu zimbra /opt/letsencrypt-zimbra/obtain-and-deploy-letsencrypt-cert.sh -vf
    ```
    `-f` is to force-renew and `-v` is for verbose, so you will see detailed output what the script is doing.

Vojtech

**Comment by CesarCapillas** [comment]
Hi Vojtech,

I answer the questions:

1. The version used is letsencrypt-zimbra version 0.3

2. Just when renewing, yes. We monitor some parts of zimbra server, the certs durability and the queues, so it seems to happen when renewing the cert.

3. Not really or not easy to see, because when experiencing problems, email server does not recieve mails (until the manual zmcontrol restart). And later, I can see some zimbra-admin messages probably of the restarting tried by the script. Anyway, the cert is renewed.

4. I will take this into consideration for debugging. Thanks for the tip.

Kind regards.
--C.

**Comment by VojtechMyslivec** [comment]
Hello,
1. The latest version is 0.4.1 (I am just preparing 0.4.2). You should keep it up-to-date, although I see no changes relevant to your issue.
2. ok
3. I see. I should implement some logging feature to the script for such cases.
4. Let me know if you discover something new
Vojtech

**Comment by CesarCapillas** [comment]
Thanks for the feedback Vojtech.
I updated the project, but I found no changes except version in shell script.

Kind regards.
--C.

**Comment by VojtechMyslivec** [comment]
I have added logging to syslog feature in #64. You can checkout the branch `syslog` and you will be able to see information messages in `/var/log/syslog` then. It will maybe helps to isolate the problem.

**Comment by VojtechMyslivec** [comment]
Closing this as the thread seems to be dead.

Feel free to reopen if you have some new info about that.
